src/components/mobile-nav-item/mobile-nav-item.tsx

```diff
@@ -0,0 +1,82 @@
+import { useState } from 'react'
+import type { FC } from 'react'
+
+import { useId } from '../../storybook/random-id'
+
+import { Icon } from '../icon'
+
+import { MobileNavItemExpandableProps, MobileNavItemSimpleProps, MobileNavSubItemProps } from './types'
+import {
+  ElMobileNavItemAnchor,
+  ElMobileNavItemBadge,
+  ElMobileNavItemContent,
+  ElMobileNavItemExpanderButton,
+  ElMobileNavItemListItem,
+  ElMobileNavItemUnorderedList,
+  ElMobileNavSubItemUnorderedList,
+} from './styles'
+import { MobileNavSubItem } from './mobile-nav-sub-item'
+
+export type MobileNavItemProps = MobileNavItemSimpleProps | MobileNavItemExpandableProps
+
+const isExpandableVariant = (props: MobileNavItemProps): props is MobileNavItemExpandableProps => {
+  return 'children' in props
+}
+
+type MobileNavItemFC = FC<MobileNavItemProps> & {
+  SubItem: FC<MobileNavSubItemProps>
+}
+
+const MobileNavItem: MobileNavItemFC = (props) => {
+  const [isExpanded, setIsExpanded] = useState(Boolean(props?.isDefaultExpanded))
+  const panelId = useId()
+
+  if (isExpandableVariant(props)) {
+    const { isActive, label, hasBadge, children } = props ?? {}
+
+    return (
+      <ElMobileNavItemUnorderedList data-is-expanded={isExpanded} role="menubar" aria-label={label}>
+        <ElMobileNavItemListItem role="none">
+          <ElMobileNavItemExpanderButton
+            type="button"
+            role="menuitem"
+            aria-expanded={isExpanded ?? isActive}
+            aria-current={isActive ? 'true' : undefined}
+            aria-controls={panelId}
+            onClick={() => setIsExpanded((prev) => !prev)}
+          >
+            <ElMobileNavItemContent>
+              {label}
+              {hasBadge && <ElMobileNavItemBadge />}
+            </ElMobileNavItemContent>
+
+            <Icon icon={isExpanded ? 'chevronUp' : 'chevronDown'} fontSize="16px" />
+          </ElMobileNavItemExpanderButton>
+
+          <ElMobileNavSubItemUnorderedList id={panelId} aria-hidden={!isExpanded} role="menu">
+            {children}
+          </ElMobileNavSubItemUnorderedList>
+        </ElMobileNavItemListItem>
+      </ElMobileNavItemUnorderedList>
+    )
+  } else {
+    const { href, isActive, label, hasBadge } = props ?? {}
+
+    return (
+      <ElMobileNavItemUnorderedList role="menubar" aria-label={label}>
+        <ElMobileNavItemListItem role="none">
+          <ElMobileNavItemAnchor href={href} aria-current={isActive ? 'page' : undefined} role="menuitem">
+            <ElMobileNavItemContent>
+              {label}
+              {hasBadge && <ElMobileNavItemBadge />}
+            </ElMobileNavItemContent>
+          </ElMobileNavItemAnchor>
+        </ElMobileNavItemListItem>
+      </ElMobileNavItemUnorderedList>
+    )
+  }
+}
+
+MobileNavItem.SubItem = MobileNavSubItem
```

Comment:
**thought:** the SubItem naming within a MobileNavItem seems odd

Reply:
**question:** do we need to have a "sub item" component or could we just allow consumers to put a MobileNavItem inside a MobileNavItem? It does mean they could have nested expandable items, but it's not clear to me if that's something we should explicitly prevent or not 🤔

Reply:
> do we need to have a "sub item" component or could we just allow consumers to put a MobileNavItem inside a MobileNavItem

technically "item"and "sub item" is similar in terms of UI, and the only reason why I declare it to be "sub item" is to prevent that scenario of having nested expandable in an expandable

> ...but it's not clear to me if that's something we should explicitly prevent or not

seems it's worthy to ask the Design for allowing the expandable in an expandable for mobile navigation menu, I'll update the conversation update for this

Reply:
Even if Design don't want to permit nested expandable items from a Design System guidelines perspective, we don't necessarily have to go to effort to prevent that from a code perspective. Engineers will be implementing designs produced by the design team, so the design team just won't produce designs that have nested expandable items. Plus, with or without `MobileNavItem.SubItem`, it will still be possible for someone to do:

```
<MobileNavItem ...>

  <MobileNavItem ...>
    <MobileNavItem.SubItem ... />
    <MobileNavItem.SubItem ... />
  </MobileNavItem ...>

  <MobileNavItem.SubItem ... />
  <MobileNavItem.SubItem ... />
</MobileNavItem ...>
```

From a guidelines perspective, this would be wrong, but its also not a costly mistake. So yeah, I'm not sure there's a whole lot of value have SubItem involved, but it's also not costing us a lot to have it, and it probably does help communicate the correct intention 🤷

Reply:
my bad, yes that's a typo